app/vmalert/alerting.go

```diff
@@ -143,6 +147,28 @@ func (ar *AlertingRule) ID() uint64 {
 	return ar.RuleID
 }
 
+func (ar *AlertingRule) debugf(at time.Time, a *notifier.Alert, format string, args ...interface{}) {
+	if !ar.Debug {
+		return
+	}
+	prefix := fmt.Sprintf("DEBUG rule %q:%q (%d) at %v: ",
+		ar.GroupName, ar.Name, ar.RuleID, at.Format(time.RFC3339))
+	if a != nil {
+		var labelKeys []string
```

Comment:
nit: we can make slice `labelKeys` with cap of a.Labels like
`labelKeys := make([]string, 0, len(a.Labels))`